components/python/pygobject-3-legacy/pygobject-3-27.p5m

```diff
@@ -0,0 +1,69 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2021 Gary Mills
+# Copyright 2016 Alexander Pyhalov
+#
+
+# This file intentionally hardcodes Python 2.7 instead of using the PYVER
+# subsititutions to avoid triggering the auto-generation of a generic
+# library/python/pygobject-3 in this component, since that's being done in
+# ../pygobject-3 instead to cover the other python 3.x versions correctly.
+
+set name=pkg.fmri value=pkg:/library/python/pygobject-3-27@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="Python bindings for GObject"
+set name=info.classification value="org.opensolaris.category.2008:Development/Python"
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license COPYING license="LGPL v2.1"
+
+depend type=require fmri=runtime/python-27
+depend type=require fmri=library/python/pygobject-3
+depend type=require fmri=library/desktop/gtk3@3.14.15,5.11-2016.0.0.2
```

Comment:
I think that depending on pygobject-3 is wrong here. We have it the other way around: pygobject-3 depends on specialized packages like pygobject-3-35 because pygobject-3 is only a meta package. If you install pygobject-3 you'll get pygobject packages for all python versions. If you install a pygobject-3 package for a certain python version, eg. pygobject-3-35 you don't necessarily want all other python versions of this package, too.

Reply:
Getting packages for all python versions is normal behavior for packages published via: *GENFRAG.p5m *PYVER.p5m .  This is our model, even though they are published from separate component directories.

However, since python-27 and -35 will soon be removed, the next push will change it to a conditional dependency.  The condition will be the presence of the appropriate python runtime package.  This change is already done for the reverse dependency.